data_ocean/converter.py

```
@@ -246,7 +246,10 @@ def process(self, start_index=0):
         if records_len:
             self.save_to_db(records)
         if start_index == 0:
-            self.delete_outdated_companies()
+            try:
+                self.delete_outdated()
+            except:
+                print('This converter has not ability to remove outdated records')
```

dont catch all exceptions

---

you must understand what error you are catching

---

also try hasattr or getattr builtins

---

refactored